neqo-transport/src/connection/mod.rs

```diff
@@ -1030,12 +1037,16 @@ impl Connection {
         }
     }
 
-    /// Process any saved packets if the appropriate keys are available.
-    fn process_saved(&mut self, cspace: CryptoSpace) {
-        if self.crypto.states.rx_hp(cspace).is_some() {
-            for saved in self.saved_datagrams.take_saved(cspace) {
-                qtrace!([self], "process saved @{:?}: {:?}", saved.t, saved.d);
-                self.process_input(saved.d, saved.t);
+    /// Process any saved datagrams that might be available for processing.
+    fn process_saved(&mut self, now: Instant) {
+        while let Some(cspace) = self.saved_datagrams.available() {
+            qdebug!([self], "process saved for space {:?}", cspace);
+            if self.crypto.states.rx_hp(cspace).is_some() {
```

Comment:
I do not like that this is theory could loop if we have mismatch between making a space available and `self.crypto.states.rx_hp(cspace).is_some()` returning false. Can we do it a bit different?

Reply:
Hmm.  I think that we have a guarantee that we won't ever mark something as available if the crypto space isn't available.  I think that what we really need here is an assertion rather than an if statement, like:

```rust
debug_assert!(self.crypto.states.rx_hp(cspace).is_some());
```

Would that make you more comfortable?

Reply:
let's do that.

Please add the assertion and land the PR.